### PR TITLE
Add Docker AIO support

### PR DESCRIPTION
**Pull request by leseb**
This commit introduces a container that runs all the Ceph service. This
is likely a step toward a common image that gets activated on demand.
Ultimately we want a single container in which we can activate which
service we want.

Signed-off-by: Sébastien Han <sebastien.han@enovance.com>

### CODE REVIEW

**Comment by Ulexus** [comment]
My issue with this is that I would call it more a demo.  It's not really a configuration that I envision being deployed anywhere but for a demo/test/dev rig.  It is what it says:  all-in-one; I'm just hesitant to put it forth... un-caveated?

**Comment by leseb** [comment]
The intended goal is mainly for people not familiar with Ceph to quickly bootstrap a complete cluster.
Actually this is something we need for the Kubernetes guys in this patch: https://github.com/GoogleCloudPlatform/kubernetes/pull/6649/files#r28358644

I'd rather prefer this container to live in this repo than anywhere else.

Sorry what do you mean "un-caveated"? :)

**Comment by Ulexus** [comment]
Sorry; I tend to fabricate terms at times.  I'm really just saying that this feels more like a demo bootstrap... something to let people get a full Ceph cluster up to play with.  It's not designed to be used in production.

I guess I'm really just complaining about the name.  I think there should be an inherent warning to keep people from trying to use this as a canonical production component.  Perhaps am I merely being protective about something that doesn't need it.

As instruction, though, (as for what I'm assuming you are wanting to use it for that PR), I think it is excellent, and I'm happy to pull it in.  I just want to call it "demo" or something, rather than "all-in-one", which, if I were looking, I would say:  "Why mess with all the other junk; let me just use this one."

All of that aside, I still think the idea of having a single container image which is configurably-executable (you can run-time select which ceph component(s) run), it great.

TLDR; don't consider my objection to be a blocker, but a request to consider its name and potential misapprehension by users.

**Comment by Ulexus** [comment]
Of course, perhaps I am missing the point, when you say bootstrap.  Is the idea here that you can create a cluster with all the necessary keys and configuration for export into a production cluster grown from it?  In that case, I would call it "bootstrap" (or similar) and provide details of its use-case in its `README.md`.

**Comment by leseb** [comment]
I agree that this is a demo. Something that can help Docker users unfamiliar with Ceph to quickly get started or test a Ceph feature.
Here I'm trying to redruce all the steps required to get a Ceph up and running.
So basically "docker run ..." would give you everything, after this just collect the keys in /etc/ceph and start doing what you want.

I can flag this container as demo if you prefer, I just thought that saying AIO would ring a bell for educated users.
I'll rename it for "demo".
I should definitely add more context in the README.

**Comment by leseb** [comment]
Just sent a new version.

**Comment by Ulexus** [comment]
Thanks; I'm much more sanguine about this, now.
